### PR TITLE
Use knapsack to set watchdog values; restart osquery on watchdog flag changes

### PR DESCRIPTION
**Pull request by RebeccaMahany**
Relates to https://github.com/kolide/launcher/issues/1224

### CODE REVIEW

File: pkg/osquery/runtime/osqueryinstance.go [hunk 1]
```diff
@@ -76,12 +76,6 @@ func WithUpdateChannel(channel string) OsqueryInstanceOption {
 	}
 }
 
-func WithEnableWatchdog(enableWatchdog bool) OsqueryInstanceOption {
```
Comment (directionless) [inline pkg/osquery/runtime/osqueryinstance.go:79]: Excellent. Ditch this for knapsack. 🚀

**Review by directionless — COMMENTED** [review]
looks pretty solid

**Review by James-Pickett — APPROVED** [review]
nice

**Review by zackattack01 — APPROVED** [review]
🚀

**Pull request merged**